tutorials/holoscan_debugging/README.md

```diff
@@ -0,0 +1,286 @@
+```
+
+The script runs through the following steps:
+1. Builds the tutorial container environment with `gdb` based on the Holoscan SDK NGC container.
+2. Builds the Endoscopy Tool Tracking application in the container environment. By default we use the
+  `debug` build mode to generate detailed debugging symbols for the Endoscopy Tool Tracking application.
+  Note that this does _not_ regenerate build symbols for Holoscan SDK, which are already packaged in
+  Holoscan SDK binaries in `rel-debug` mode.
+3. Launches the Endoscopy Tool Tracking application with `gdb`. This step prefixes the launch command
+  given by `./run launch endoscopy_tool_tracking` to run with `gdb`. The command sets a few actions for `gdb` to take on startup:
+    - Sets a breakpoint in the `main` function of Endoscopy Tool Tracking;
+    - Runs the program with custom arguments until the breakpoint is hit;
+    - Sets a breakpoint in Holoscan SDK's `add_flow` function.
+
+At this point `gdb` enters an interactive session where we can inspect the Endoscopy Tool Tracking program state and advance execution.
+
+From this point we recommend referring to the [GDB Manual](https://sourceware.org/gdb/current/onlinedocs/gdb.html/) or
+online tutorials to get started with interactive debugging commands.
+
+### Frequently Asked Questions and Troubleshooting
+
+#### How can I verify that Holoscan SDK debugging symbols have loaded in `gdb`?
+
+`gdb` loads debugging symbols for Holoscan SDK only when the application loads Holoscan SDK binaries. Before that time, we can set breakpoints in Holoscan SDK files, but `gdb` will not understand them yet.
+We can use `info sources` to inspect that Holoscan SDK symbols are available.
+
+```bash
+(gdb) info sources /workspace/holoscan-sdk/src/core
+/workspace/holoscan-sdk/src/core/application.cpp, /workspace/holoscan-sdk/src/core/services/generated/system_resource.grpc.pb.cc,
+/workspace/holoscan-sdk/src/core/services/generated/system_resource.pb.h, /workspace/holoscan-sdk/src/core/services/generated/system_resource.pb.cc,
+/workspace/holoscan-sdk/src/core/services/generated/result.grpc.pb.cc, /workspace/holoscan-sdk/src/core/services/generated/result.pb.h,
+...
+```
+
+#### How can I attach to a running Holoscan SDK session?
+
+Do the following to attach to a HoloHub application:
+
+1. Launch the container with root permissions and start the process in the background:
+```bash
+./dev_container launch --img holohub:debugging --as_root
+>>> ./run launch endoscopy_tool_tracking &     # then Ctrl+C to return to your interactive shell
+```
+
+2. Find the process ID (PID) of the running application:
+```bash
+>>> ps -ef | grep endoscopy_tool_tracking
+user+     292     203 28 13:17 pts/9    00:00:04 /workspace/holohub/build/endoscopy_tool_tracking/applications/endoscopy_tool_tracking/cpp/endoscopy_tool_tracking --data /workspace/holohub/data/endoscopy
+```
+
+3. Attach to the running process with `gdb -p`:
+```bash
+>>> gdb -p 292
```

Comment:
I got the following error, and I assume that's because we don't include GXF extension debug symbols. Do we want to add a note to this?

```bash
warning: Could not load shared library symbols for gxf_extensions/lstm_tensor_rt_inference/libgxf_lstm_tensor_rt_inference.so.
```

Reply:
Fixed, this was an issue with the working directory. Thanks for reporting